core/src/main/scala/org/apache/spark/deploy/security/HBaseDelegationTokenProvider.scala

```diff
@@ -70,6 +70,10 @@ private[security] class HBaseDelegationTokenProvider
         getMethod("create", classOf[Configuration])
       confCreate.invoke(null, conf).asInstanceOf[Configuration]
     } catch {
+      case e: ClassNotFoundException =>
+        logWarning(s"You are attempting to use the ${getClass.getCanonicalName}, but your" +
```

Comment:
Unlike Hadoop and Hive, Spark will not be built with HBase libs. It will only work when end users link the HBase libraries or contain HBase libraries in their app jar.

So the message doesn't look correct, though I think `Fail to invoke HBaseConfiguration` is much misleading.

Actually the root issue is that HBaseDelegationToken is activated even end users don't use HBase in their apps, which #23499 is trying to avoid with service loader.

Reply:
Thanks I will check the PR